trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1082,8 +1094,85 @@ where
                 self.process_content(content, source, find_content_request)
                     .await
             }
-            _ => {}
+            Response::Accept(accept) => {
+                let offer_request = match request {
+                    Request::Offer(offer) => offer,
+                    _ => {
+                        error!("Unable to process received content: Invalid request message.");
+                        return;
+                    }
+                };
+
+                if let Err(err) = self
+                    .process_accept(accept, source, offer_request.content_keys)
+                    .await
+                {
+                    error!("Error processing ACCEPT response in overlay service: {err}")
+                }
+            }
+        }
+    }
+
+    /// Process ACCEPT response
+    pub async fn process_accept(
+        &self,
+        response: Accept,
+        enr: Enr,
+        content_keys: Vec<RawContentKey>,
+    ) -> anyhow::Result<Accept> {
+        let content_keys_offered: Result<Vec<TContentKey>, TContentKey::Error> = content_keys
+            .into_iter()
+            .map(|key| (TContentKey::try_from)(key))
+            .collect();
+
+        let content_keys_offered: Vec<TContentKey> = match content_keys_offered {
+            Ok(val) => val,
+            Err(_) => {
+                return Err(anyhow!("Unable to decode offered content keys"));
```

Comment:
Just to highlight how weird this is:
```suggestion
                return Err(anyhow!("Unable to decode our own offered content keys"));
```